Example/ReactiveDataDisplayManager/Table/TwoDirectionPaginatableTableViewController/TwoDirectionPaginatableTableViewController.swift

```diff
@@ -0,0 +1,214 @@
+    private lazy var backwardPaginationDelegate = BackwardPaginationDelegate(input: self, prevPageAction: loadPrevPage)
+
+    private var isFirstPageLoading = true
+    private var currentPage = 0
+
+    // MARK: - UIViewController
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Table with two directions pagination"
+
+        configureActivityIndicatorIfNeeded()
+        loadFirstPage()
+    }
+
+    // MARK: - Methods
+
+    func initializePaginationInput(input: PaginatableInput) {
+        if isTableBackwardPaginationInput(input) {
+            backwardPaginatableInput = input
+        } else {
+            forwardPaginatableInput = input
+        }
+    }
+
+}
+
+// MARK: - Configuration
+
+private extension TwoDirectionPaginatableTableViewController {
+
+    func configureActivityIndicatorIfNeeded() {
+        if #available(iOS 13.0, tvOS 13.0, *) {
+            activityIndicator.style = .medium
+        }
+    }
+
+}
+
+// MARK: - Private Methods
+
+private extension TwoDirectionPaginatableTableViewController {
+
+    func loadFirstPage() {
+
+        // show loader
+        activityIndicator.isHidden = false
+        activityIndicator.startAnimating()
+
+        // hide footer and header
+        forwardPaginatableInput?.updatePagination(canIterate: false)
+        backwardPaginatableInput?.updatePagination(canIterate: false)
+
+        // imitation of loading first page
+        delay(.now() + .seconds(1)) { [weak self] in
+
+            // fill table
+            self?.fillAdapter()
+
+            // scroll to middle
+            self?.tableView.scrollToRow(at: Constants.firstPageMiddleIndexPath, at: .middle, animated: false)
+
+            // hide loader
+            self?.activityIndicator?.stopAnimating()
+            self?.activityIndicator?.isHidden = true
+
+            // show pagination loader if update is needed
+            self?.forwardPaginatableInput?.updatePagination(canIterate: true)
+            self?.backwardPaginatableInput?.updatePagination(canIterate: true)
+        }
+    }
+
+    /// This method is used to fill adapter
+    func fillAdapter() {
+
+        for _ in 0...Constants.pageSize {
+            adapter += makeGenerator()
+        }
+
+        adapter => .reload
+    }
+
+    func makeGenerator() -> TableCellGenerator {
+        TitleTableViewCell.rddm.baseGenerator(with: "Random cell \(Int.random(in: 0...1000)) from page \(currentPage)" )
+    }
+
+    func canFillPages() -> Bool {
+        if isFirstPageLoading {
+            isFirstPageLoading.toggle()
+            return false
+        } else {
+            return true
+        }
+    }
+
+    func fillNext() -> Bool {
+        currentPage += 1
+
+        for _ in 0...Constants.pageSize {
+            adapter += makeGenerator()
+        }
+
+        adapter => .reload
+
+        return currentPage < Constants.pagesCount
+    }
+
+    func fillPrev() -> Bool {
+        currentPage += 1
```

Comment:
```suggestion
        currentPage -= 1
```
по логике так должно быть)

Reply:
Я думал ограничить пагинацию 10-ю страницами, не важно в какую сторону. Но этот вариант тоже норм, чтобы ввести currentPage в ноль, надо все равно попробовать пролистать и вперед и назад)